core/src/main/java/org/keycloak/jose/jwk/JWKBuilder.java

```diff
@@ -77,6 +78,13 @@ public JWK rsa(Key key) {
         return k;
     }
 
+    public JWK rsa(Key key, KeyUse keyUse) {
+        JWK k = rsa(key);
+        String keyUseString = DEFAULT_PUBLIC_KEY_USE;
+        if (KeyUse.ENC == keyUse) keyUseString = "enc";
```

Comment:
According to https://tools.ietf.org/html/rfc7517#section-4.2 , "sig" and "enc" are specified while org.keycloak.crypto.KeyUse specifies "SIG" and "ENC".

To comply with JWK's specification, I've not used KeyUse.toString() directly.

How do you think about the following fix?
`String keyUseString = keyUse == null ? DEFAULT_PUBLIC_KEY_USE : keyUse.toLowerCase();`

IMO, it might be better that keycloak's internal constants be the same as corresponding specification's representation. (e.g., JWK's "sig", "enc")

Reply:
Regarding to what you mentioned, the java constants are usually upper-case. However the option could be to refactor the KeyUse class like this:
```
public enum KeyUse {
    
    SIG("sig"),
    ENC("enc");
    
    private String specName;
    
    KeyUse(String specName) {
        this.specName = specName;
    }

    public String getSpecName() {
        return specName;
    }
}
```

Then you can "getSpecName()" . But if you prefer to not change it this way in this PR, I am fine with the "toLowerCase" as well.